coral/src/app/features/topics/details/components/PromotionBanner.tsx

```diff
@@ -0,0 +1,88 @@
+import { Banner, Box, Button, GridItem } from "@aivenio/aquarium";
+import illustration from "src/app/images/topic-details-schema-Illustration.svg";
+import { ReactElement } from "react";
+import { KlawApiModel } from "types/utils";
+
+interface PromotionBannerProps {
+  // `topicName` is only optional on
+  // KlawApiModel<"PromotionStatus">
+  // so we can't rely on it to be part of
+  // the promotionDetails
+  topicName: string;
+  promotionDetails: KlawApiModel<"PromotionStatus">;
```

Comment:
Should `KlawApiModel<"PromotionStatus">` be aliased to a type in `domain` so we don't expose `KlawApiModel` in `app`?

Reply:
Yes, that's a good idea! 

Should we have a directory `promotion` in `domain` 🤔  We've that for `requests`, too.

Reply:
Yes, but there would only be `types` currently, right? we want to keep `promote{entity` in the related entity domain?

Reply:
Yes!

Reply:
Done! https://github.com/aiven/klaw/pull/1513/commits/a256e8be4b701f99cdd868fdfcc31aa7d1bd167a